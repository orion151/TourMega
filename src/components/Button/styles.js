import {
  StyleSheet
} from 'react-native'

import { BLACK_COLOR, WHITE_COLOR } from '../../constants'

const styles = StyleSheet.create({
  buttonContainer: {
    width: 150,
    height: 50,
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 10,
    backgroundColor: BLACK_COLOR
  },
  buttonLabel: {
    color: WHITE_COLOR,
    fontSize: 18
  }
})

export default styles
