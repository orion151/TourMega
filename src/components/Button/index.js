import React from 'react'

import {
  View,
  Text,
  TouchableOpacity
} from 'react-native'

import PropTypes from 'prop-types'

import styles from './styles'

const Button = (props) => {
  const { onClick } = props

  return (
      <TouchableOpacity onPress={() => onClick()}>
        <View style={styles.buttonContainer}>
            <Text style={styles.buttonLabel}>View Details</Text>
        </View>
      </TouchableOpacity>
  )
}

Button.propTypes = {
  onClick: PropTypes.func.isRequired
}

export default Button
