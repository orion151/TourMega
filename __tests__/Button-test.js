import { expect } from '@jest/globals'
import React from 'react'
import renderer from 'react-test-renderer'

import Button from '../src/components/Button/index'

test('tenders correctly', () => {
  const tree = renderer.create(<Button />).toJSON()
  expect(tree).toMatchSnapshot()
})
