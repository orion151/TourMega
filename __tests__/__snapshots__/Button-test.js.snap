// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tenders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "width": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 18,
        }
      }
    >
      View Details
    </Text>
  </View>
</View>
`;
